sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp

```diff
@@ -107,6 +107,11 @@ namespace Azure { namespace Storage { namespace Blobs {
      * @brief Specify the number of retries and other retry-related options.
      */
     StorageRetryWithSecondaryOptions RetryOptions;
+
+    /**
+     * @brief Customized HTTP client. We're going to use the default one if this is empty.
+     */
+    std::shared_ptr<Core::Http::HttpTransport> TransportAdapter;
```

Comment:
This is expected to be the `TransportPolicyOptions` structure over here. That structure contains the TransportAdapter.
The end user would do:
```cpp
StorageBlobClientOptions options;
options.TransportPolicyOptions.TransportAdapter = `end user transport adapter if they want`;
StorageClient client(url, credential, options);
```

In the above code, end user can omit adding the client options and then Azure Core will add the default.
Or also, the end user can pass the client options but without setting anything for the TransportAdapterOptions (they could only set the retryPolicyOptions, for instance) and Azure Core would also add the defaults.

Reply:
you guys have made these stuff too complex for me to grasp.